### PR TITLE
Edited javadoc of checkAttributeSemantics

### PR DESCRIPTION
**Pull request by metodej**
- edited thrown exceptions in checkAttributeSyntax methods
- also fixed some bugs where checkAttributeSyntax was not called (in virtual attributes googleGroupsName and unixGroupName)

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/FacilityAttributesModuleImplApi.java [hunk 1]
```diff
@@ -36,13 +36,11 @@
 	 * @param attribute attribute to check
 	 * @throws InternalErrorException if an exception is raised in particular
 	 *         implementation, the exception is wrapped in InternalErrorException
-	 * @throws WrongAttributeValueException if the attribute value is wrong/illegal
-	 * @throws WrongReferenceAttributeValueException if an referenced attribute against
-	 *         the parameter is to be compared is not available
+	 * @throws WrongReferenceAttributeValueException if the attribute value is wrong/illegal semantics
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/FacilityAttributesModuleImplApi.java]: is -> has

File: perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/UserFacilityAttributesModuleImplApi.java [hunk 2]
```diff
@@ -39,12 +39,10 @@
 	 * @param attribute Attribute in relationship between facility and user.
 	 * @throws InternalErrorException if an exception is raised in particular
 	 *         implementation, the exception is wrapped in InternalErrorException
-	 * @throws WrongAttributeValueException if the attribute value is wrong/illegal
-	 * @throws WrongReferenceAttributeValueException if an referenced attribute against
-	 *         the parameter is to be compared is not available
+	 * @throws WrongReferenceAttributeValueException if the attribute value is wrong/illegal semantics
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/attributes/UserFacilityAttributesModuleImplApi.java]: is -> has

File: perun-core/src/main/java/cz/metacentrum/perun/core/bl/AttributesManagerBl.java [hunk 3]
```diff
@@ -2768,17 +2764,16 @@
...
-	 * @throws WrongAttributeValueException if the attribute value is wrong/illegal
 	 * @throws WrongAttributeAssignmentException if attribute isn't member-resource attribute
-	 * @throws WrongReferenceAttributeValueException
+	 * @throws WrongReferenceAttributeValueException if the attribute value has wrong/illegal semantics
 	 */
-	void checkAttributeSemantics(PerunSession sess, Member member, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongAttributeAssignmentException, WrongReferenceAttributeValueException;
+	void checkAttributeSemantics(PerunSession sess, Member member, Attribute attribute) throws InternalErrorException, WrongAttributeAssignmentException, WrongReferenceAttributeValueException;
 
 	/**
 	 *  Batch version of fillAttribute
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/bl/AttributesManagerBl.java]: Not your change, but can you please either update this or remove it. It was probably copy-pasted from fillAttribute().

**Comment by metodej** [comment]
Thanks for review, it should be fixed now and I will add PR for the two mentioned classes shortly.

**Pull request merged**